ibis/expr/tests/test_api.py

```diff
@@ -147,3 +147,12 @@ def test_implicit_coercion_of_null_literal(op):
 
     assert expr1.op() == expected
     assert expr2.op() == expected
+
+
+def test_nested_name_property():
+    x = ibis.literal(1)
+    n = 100
+    for _ in range(n):  # noqa: F402
+        x = x + 1
+
+    assert x.op().name.count("Add") == n
```

Comment:
@cpcloud do we want to add some way of checking for recursion explosion? Maybe use n=200 and then test this doesn't take longer than 3 seconds to compute?

Reply:
It didn't finish in any reasonable amount of time, so just having it as a standard unit test seems fine.

Reply:
ahh, I see, so you would just notice it during local dev, no automation needed. Sure that works.